test/test_model.py

```diff
@@ -1484,87 +1607,202 @@ def test_unit_get_container(self):
+        self.requests.append(('push', path, source, encoding, make_dirs, permissions,
+                              user_id, user, group_id, group))
+
+    def list_files(self, path: str, *, pattern: typing.Optional[str] = None, itself: bool = False):
+        self.requests.append(('list_files', path, pattern, itself))
+        return self.responses.pop(0)
+
+    def make_dir(
+            self,
+            path: str,
+            *,
+            make_parents: bool = False,
+            permissions: typing.Optional[int] = None,
+            user_id: typing.Optional[int] = None,
+            user: typing.Optional[str] = None,
+            group_id: typing.Optional[int] = None,
+            group: typing.Optional[str] = None):
+        self.requests.append(('make_dir', path, make_parents, permissions, user_id, user,
+                              group_id, group))
+
+    def remove_path(self, path: str, *, recursive: bool = False):
+        self.requests.append(('remove_path', path, recursive))
+
+    def exec(self, command: typing.List[str], **kwargs: typing.Any):
+        self.requests.append(('exec', command, kwargs))
+        return self.responses.pop(0)
+
+    def send_signal(self, signal: typing.Union[str, int], service_names: str):
+        self.requests.append(('send_signal', signal, service_names))
+
+    def get_notice(self, id: str) -> pebble.Notice:
+        self.requests.append(('get_notice', id))
+        return self.responses.pop(0)
+
+    def get_notices(self, **kwargs: typing.Any):
+        self.requests.append(('get_notices', kwargs))
+        return self.responses.pop(0)
+
+
+class TestContainerPebble:
+    @pytest.fixture
+    def model(self):
         meta = ops.CharmMeta.from_yaml("""
 name: k8s-charm
 containers:
   c1:
     k: v
 """)
         backend = MockPebbleBackend('myapp/0')
-        self.model = ops.Model(meta, backend)
-        self.container = self.model.unit.containers['c1']
-        self.pebble: MockPebbleClient = self.container.pebble  # type: ignore
-
-    def test_socket_path(self):
-        assert self.pebble.socket_path == '/charm/containers/c1/pebble.socket'
-
-    def test_autostart(self):
-        self.container.autostart()
-        assert self.pebble.requests == [('autostart',)]
-
-    def test_replan(self):
-        self.container.replan()
-        assert self.pebble.requests == [('replan',)]
-
-    def test_can_connect(self):
-        self.pebble.responses.append(pebble.SystemInfo.from_dict({'version': '1.0.0'}))
-        assert self.container.can_connect()
-        assert self.pebble.requests == [('get_system_info',)]
-
-    def test_start(self):
-        self.container.start('foo')
-        self.container.start('foo', 'bar')
-        assert self.pebble.requests == [
+        return ops.Model(meta, backend)
+
+    @pytest.fixture
+    def container(self, model: ops.Model):
+        return model.unit.containers['c1']
+
+    @pytest.fixture
+    def container_pebble(self, container: ops.Container):
```

Comment:
Here too, I'd much rather see a single fixture instead of the nested dependent fixtures that you have to trace up. It looks like it'd make sense to have a single `container` fixture here that creates the model and return the c1 container. And the tests (which don't seem to use `model` directly anyway) would reference `container.pebble` explicitly (that's no longer than `container_pebble`).